Create guide about generating a genesis block - Closes #942 (#1080)
LiskHQ/lisk-docs: modules/ROOT/pages/guides/app-development/genesis-block.adoc

Beside this, any other accounts can be added here, and account properties like `balance` can be configured as desired.

[source,js]
----
// Creating the genesis account list

const newAccount = () => {
  const cred = newCredentials();
  credentials.push(cred);
  const account = {
    address: Buffer.from(cred.binaryAddress, 'hex'),
    token: { balance: BigInt(25000000000) }
  };
  return account;
};

const generateAccounts = (amount) => {
  const accounts = [];
  for (let i = 1; i <= amount; i++) {
    accounts.push(newAccount())
  }
  return accounts;
};

const genesisAccounts = generateAccounts(3);

const accounts = [...delegates, ...genesisAccounts];
----

== Creating the genesis block

As final step, use the function `createGenesisBlock()` of the genesis library to generate the genesis block.

The parameters for the genesis block are all packed into one object.

* `initDelegates`(Array): List of initial delegate addresses used during the bootstrap period to forge blocks.
Addresses are expected to be in `Buffer` format.
* `accounts`(Array): List of initial accounts in the network.

> Minimum required is `address`, but can have other properties like `balance`.

Reviewer: ```suggestion
The minimum required is `address`, however other properties such as `balance` can be included.
```

Addresses are expected to be in `Buffer` format.
* `accountAssetSchemas`(Object): The genesis block needs to contain all account asset schemas for all modules which are registered with the respective blockchain application.

> The different account asset schemas are all grouped together in one big object and added as `accountAssetSchemas` to the genesis block params.

Reviewer: ```suggestion
The different account asset schemas are all grouped together in one large object and added as `accountAssetSchemas` to the genesis block params.
```

`accountAssetSchemas` is one of the most important parameters for generating a valid genesis block, so make sure it includes all required account asset schemas.

> This object is provided as parameter for the `createGenesisBlock()` function, which will then use them to generate the dedicated genesis block.

Reviewer: ```suggestion
This object is provided as a parameter for the `createGenesisBlock()` function, which will then be used to generate the dedicated genesis block.
```

    token: { balance: BigInt(25000000000) }
  };
  return account;
};

const generateAccounts = (amount) => {
  const accounts = [];
  for (let i = 1; i <= amount; i++) {
    accounts.push(newAccount())
  }
  return accounts;
};

const genesisAccounts = generateAccounts(3);

const accounts = [...delegates, ...genesisAccounts];

// Creating the genesis block

const genesisBlockParams = {
  initDelegates: delegates.map(a => a.address),
  accounts,
  accountAssetSchemas,
};

const genesisBlock = genesis.createGenesisBlock(genesisBlockParams);

console.log(genesisBlock);

----

.Example output
[source,js]
----
{
  header: {
    generatorPublicKey: <Buffer >,
    height: 0,
    previousBlockID: <Buffer >,
    reward: 0n,
    signature: <Buffer >,
    timestamp: 1620398241,
    transactionRoot: <Buffer e3 b0 c4 42 98 fc 1c 14 9a fb f4 c8 99 6f b9 24 27 ae 41 e4 64 9b 93 4c a4 95 99 1b 78 52 b8 55>,
    version: 0,
    asset: { initRounds: 3, initDelegates: [Array], accounts: [Array] },
    id: <Buffer a8 eb 2c da 7b 2a 8f 0e 84 24 1e 78 af 78 4e c9 0b ce d0 39 d0 b4 98 87 95 a8 5d 63 83 be c5 a0>
  },
  payload: []
}
----

== Optional parameters for the genesis block

> Beside the three required properties `initDelegates`, `accounts`, and `accountAssetSchemas`, it is possible to set the following optional propeerties for the genesis block:

Reviewer: ```suggestion
Besides the three required properties `initDelegates`, `accounts`, and `accountAssetSchemas`, it is possible to set the following optional properties for the genesis block:
```
